src/robot/ConcreteRobot.cpp

```diff
@@ -245,16 +264,34 @@ TrajectoryPtr ConcreteRobot::planToConfiguration(
     const CollisionFreePtr& collisionFree,
     double timelimit)
 {
+  DART_UNUSED(timelimit);
+
+  auto robot = metaSkeleton->getBodyNode(0)->getSkeleton();
+  std::lock_guard<std::mutex> lock(robot->getMutex());
+  // Save the current state of the space.
+  auto saver = MetaSkeletonStateSaver(metaSkeleton);
+  DART_UNUSED(saver);
+
+  auto snapConfigToConfigPlanner
+      = std::make_shared<SnapConfigurationToConfigurationPlanner>(
+          stateSpace, std::make_shared<GeodesicInterpolator>(stateSpace));
+
+  // Convert to DART planner.
+  auto dartSnapConfigToConfigPlanner = std::make_shared<
+      ConfigurationToConfiguration_to_ConfigurationToConfiguration>(
+      snapConfigToConfigPlanner, metaSkeleton);
+
+  // Create planning problem.
+  auto castedGoalState
+      = static_cast<const statespace::dart::MetaSkeletonStateSpace::State*>(
```

Comment:
For some reason, I thought that we take the goal state as an `Eigen::VectorXd` instead. Is there a reason we take a `State`? What happens if this isn't actually a `MetaSkeletonStateSpace::State`?

Reply:
Yeah, this is something I noticed as well. I'm not sure why we decided to take `StateSpace::State*` instead of `MetaSkeletonStateSpace::State*`. Since we're working with `ConcreteRobot` we must be working with `MetaSkeletonStateSpace::State*` (unless the user is doing something very stupid).

I agree that this design decision is a little funky (especially considering that we take a `metaSkeleton` as another argument), but changing it would break a lot of things just because of type issues. Maybe we can file an issue to clean this up later?

Reply:
@aditya-vk does this seem ok for now?